File: Generacidad/Ej1/Caja.py
class Caja:
    def __init__(self):
        self.contenido = None

    def guardar(self, item):
        self.contenido = item

    def obtener(self):
        return self.contenido


caja_texto = Caja()
caja_texto.guardar("¡Hola mundo!")
print(f"Caja de texto: {caja_texto.obtener()}")

caja_numero = Caja()
caja_numero.guardar(123)
print(f"Caja de número: {caja_numero.obtener()}")

File: Generacidad/Ej1/Caja.java
package Generacidad.Ej1;

public class Caja<T> {
    private T contenido;

    public void guardar(T item) {
        this.contenido = item;
    }

    public T obtener() {
        return contenido;
    }

    public static void main(String[] args) {
        Caja<String> cajaTexto = new Caja<>();
        cajaTexto.guardar("¡Hola mundo!");
        System.out.println("Caja de texto: " + cajaTexto.obtener());

        Caja<Integer> cajaNumero = new Caja<>();
        cajaNumero.guardar(123);
        System.out.println("Caja de número: " + cajaNumero.obtener());
    }
}
